src/components/loaders/LayersLoader.js

```diff
@@ -1,29 +1,44 @@
 import React, { useEffect } from 'react'
 import { useSelector, useDispatch } from 'react-redux'
 import { setLayerLoading, updateLayer } from '../../actions/layers.js'
+import { EVENT_LAYER } from '../../constants/layers.js'
 import LayerLoader from './LayerLoader.js'
 
 const LayersLoader = () => {
-    const layers = useSelector((state) =>
-        state.map.mapViews.filter((layer) => {
-            // The layer is currently being loaded - don't load again
-            if (layer.isLoading) {
-                return false
-            } else {
-                // The layer is not loaded - load it
-                if (!layer.isLoaded) {
-                    return true
+    const layers = useSelector(({ map, dataTable }) =>
+        map.mapViews.filter(
+            ({
+                id,
+                isLoading,
+                isLoaded,
+                isExtended,
+                layer: layerType,
+                serverCluster,
+            }) => {
+                // The layer is currently being loaded - don't load again
+                if (isLoading) {
+                    return false
+                } else {
+                    // The layer is not loaded - load it
+                    if (!isLoaded) {
+                        return true
+                    }
+
+                    // The layer is loaded but the data table is now displayed and
+                    // event extended data hasn't been loaded yet - so load it
+                    if (
+                        layerType === EVENT_LAYER &&
+                        id === dataTable &&
+                        !isExtended &&
+                        !serverCluster
+                    ) {
+                        return true
+                    }
+
+                    return false
```

Comment:
Instead of explicitly returning `true` or `false` you could just return what is in the if-clause directly:
```suggestion
                    return (
                        layerType === EVENT_LAYER &&
                        id === dataTable &&
                        !isExtended &&
                        !serverCluster
                    )
```

Reply:
Code was written in an explicit way on purpose.